platform/core.startup/src/org/netbeans/core/startup/TopLogging.java

```diff
@@ -401,41 +401,47 @@ static void exit(int exit) {
         TopSecurityManager.exit(exit);
     }
 
-    private static final class LookupDel extends Handler
-    implements LookupListener {
-        private Lookup.Result<Handler> handlers;
+    static void exit(int exit, Throwable t) {
+        t.printStackTrace(OLD_ERR);
+        exit(exit);
+    }
+
+    private static final class LookupDel extends Handler {
+        private final Lookup.Result<Handler> handlers;
         private Collection<? extends Handler> instances;
 
 
-        public LookupDel() {
+        LookupDel() {
             handlers = Lookup.getDefault().lookupResult(Handler.class);
-            instances = handlers.allInstances();
-            instances.size(); // initialize
-            handlers.addLookupListener(this);
+            LookupListener onChange = (__) -> {
+                instances = new ArrayList<>(handlers.allInstances());
```

Comment:
Very tricky !
I'd suggest to add some non-normative warning to Open API docs so dumb Lookup users as myself are aware that the collection "that never changes contents" MAY be lazy computed, so it may call out on 1st content access.

Reply:
Added in 
acbe6ad ... it has been a long time since I touched the `Lookup` documentation.